Community/Tdarr_Plugin_TN10_SUBS.js

```diff
@@ -0,0 +1,193 @@
+    // eslint-disable-next-line no-console
+    response.infoLog += '☒File is not video \n';
+    response.processFile = false;
+    return response;
+  }
+
+  // Make sure file has subtitles.
+  let hasSubs = false;
+  for (let i = 0; i < file.ffProbeData.streams.length; i++) {
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_type.toLowerCase() === 'subtitle'
+      ) {
+        hasSubs = true;
+      }
+    } catch (err) {}
+  }
+
+  if (hasSubs === false) {
+    response.infoLog += '☒ No subs in file, skipping!\n';
+    response.processFile = false;
+    return response;
+  } else {
+    response.infoLog += '☑ Found subs!\n';
+  }
+
+  // Set global variables.
+  const subsArr = file.ffProbeData.streams.filter((row) => row.codec_name === 'subrip');
+  const {
+    originalLibraryFile
+  } = otherArguments;
+  let extract_lang = inputs.extract_lang.split(',');
+  let subsFile = '';
+  let subsFile_forced = '';
+  let lang = '';
+  let title = '';
+  let forced = false;
+
+  let command = '-y,';
+  try {
+    for (let i = 0; i < subsArr.length; i += 1) {
+
+      // Set per-stream variables
+      const subStream = subsArr[i];
+      lang = subStream.tags.language;
+      title = subStream.tags.title;
+      if (subStream.disposition.forced || (title.toLowerCase().includes('forced'))) {
+        forced = true;
+      }
+
+      // Build subtitle file names.
+      // for Tdarr V2 (2.00.05+)
+      if (originalLibraryFile && originalLibraryFile.file) {
+        subsFile = originalLibraryFile.file;
+        subsFile_forced = originalLibraryFile.file;
+      } else {
+        // for Tdarr V1
+        subsFile = file.file;
+        subsFile_forced = file.file;
+      }
+      subsFile = subsFile.split('.');
+      subsFile[subsFile.length - 2] += `.${lang}`;
+      subsFile[subsFile.length - 1] = 'srt';
+      subsFile = subsFile.join('.');
+      subsFile_forced = subsFile_forced.split('.');
+      subsFile_forced[subsFile_forced.length - 2] += `.${lang}.forced`;
+      subsFile_forced[subsFile_forced.length - 1] = 'srt';
+      subsFile_forced = subsFile_forced.join('.');
+
+      // Verify subtitle track is a format that can be extracted.
+      if (subsArr.length !== 0) {
+        const {
+          index
+        } = subStream;
+        // Skip commentary tracks.
+        if (title.toLowerCase().includes('commentary') || title.toLowerCase().includes('description') || subStream.disposition.comment) {
+          response.infoLog += `☒ Stream ${i} ${lang}.srt is a ${title} track. Skipping!\n`;
+          // Skip undesired languages.
+        } else if ((inputs.extract_lang !== '') && (extract_lang.indexOf(lang) === -1)) {
+          response.infoLog += `☒ Stream ${i} ${lang}.srt is an undesired language. Skipping!\n`;
+        } else {
+          // Extract subtitle if it doesn't exist on disk or the option to overwrite is set.
+          if (fs.existsSync(`${subsFile}`) && !inputs.overwrite_subs && !forced) {
```

Comment:
You don't need string interpolation here

Reply:
Fixed with next pull

Reply:
Tdarr seems to error on this if I don't use string interpolation here.